src/QUICSocket.ts

```diff
@@ -79,71 +119,107 @@ class QUICSocket extends EventTarget {
+      // https://www.rfc-editor.org/rfc/rfc9000.html#stateless-reset
+      await connection.recv(data, remoteInfo_);
+      // Failing this is a software error (not a caller error)
+    } else {
+      // If the server is not registered, we cannot attempt to create a new
+      // connection for this packet.
       if (this.server == null) {
         return;
       }
-      // At this point, the connection may not yet be started
-      const connection_ = await this.server.connectionNew(
-        remoteInfo_,
-        header,
-        dcid,
-        data,
-      );
-      // If there's no connection yet
-      // then the server is middle of version negotiation or stateless retry
-      if (connection_ == null) {
-        return;
+      try {
+        // This call will block until the connection is started which
+        // may require multiple `recv` and `send` pairs to process the
+        // received packets.
+        // In order to do this, firstly the initial `data` is faciliated by the
+        // `QUICServer`. And subsequently multiple `recv` and `send` pairs will
+        // occur concurrently while the the connection is starting.
+        // These concurrent `recv` and `send` pairs occur in this same handler,
+        // but just in the other branch of the current `if` statement where
+        // the connection object already exists in the connection map.
+        await this.server.acceptConnection(
+          remoteInfo_,
+          header,
+          dcid,
+          data,
+        );
+      } catch (e) {
+        // The acceptConnection is a caller error
+        // Now we have to handle this and decide
+        // Whether this is to be dropped
+        // OR that it's a domain error
+
+        // If the connection timed out during start, this is an expected
+        // possibility, because the remote peer might have become unavailable,
+        // in which case we can just ignore the error here.
+        if (
+          errorsUtils.checkError(
+            e,
+            (e) => e instanceof errors.ErrorQUICConnectionStartTimeout,
+          )
+        ) {
+          // This is a legitimate state transition
+          // The caller error is ignored
+          return;
+        }
+        // Translate the caller error to a domain error
+        if (
+          errorsUtils.checkError(e, (e) => e instanceof errors.ErrorQUICSocket)
+        ) {
+          // This would mean the `this.send` call somewhere failed
+          const e_ = new errors.ErrorQUICSocketInternal(
+            'Failed to call accept connection due to socket send',
+            { cause: e }
+          );
+          this.dispatchEvent(
+            new events.EventQUICSocketError({
+              detail: e_
+            })
+          );
+          this.dispatchEvent(
+            new events.EventQUICSocketClose()
+          );
+          return;
+        }
+
+        // Software error - throw upwards
+        throw e;
```

Comment:
If the server connection rejects the client TLS with custom TLS, the `acceptConnection` throws `ErrorQUICServerNewConnection: Failed to start accepted connection`.  In fact, all connection start errors are rethown as `ErrorQUICServerNewConnection: Failed to start accepted connection`. In this case this error should be ignored here as it's a recoverable error from the server and socket perspective.

Reply:
The `checkError` checks recursively.

Reply:
Sure, but `ErrorQUICConnectionStartTimeout` is not the only error we want to ignore here. There's timeout from `maxIdleTimeout`, abortion, TLS failure, etc etc..

Reply:
Right so we should make a list or just check for new connection failure in general?

Reply:
Done in 67c28f67555ad53ab4bd7d19cc5cdd50407cdbb0.